src/core/flow.js

```diff
@@ -101,18 +101,22 @@ export function Flow({ ctx, events, init, validator, processMove }) {
  *                               Whenever `condition` is true the `action` is run.
  *                               Triggers are processed one after the other in the
  *                               order they are defined at the end of each move.
+ * @param {...object} allowEndTurnIf - Turn not allowed to end if this returns true.
```

Comment:
Hmm, I think this is going to make the semantics a bit confusing. Like you mention, what happens if this conflicts with `endTurnIf`?

How about just using a simple whitelist similar to `allowedMoves`. In the flow section, you can have:

```
flow: {
  // default value if not provided.
  allowedEvents: [ 'endTurn', 'endPhase', 'pass'],
}
```

Reply:
Yes the conflict with endTurnIf is a problem. Simply whitelisting would work.
For my use case the 'endTurn' event is only allowed after specific moves have been made,
but I can make it work by (a) disabling it using allowedEvents on flow and (b) implementing a move which sets a flag on game and (c) endTurnIf picks up the flag.